miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/RealMiniAppDisplay.kt

```diff
@@ -1,97 +1,57 @@
 package com.rakuten.tech.mobile.miniapp.display
 
 import android.annotation.SuppressLint
+import android.app.Activity
 import android.content.Context
 import android.view.View
-import android.view.ViewGroup
-import android.webkit.WebView
-import android.widget.FrameLayout
 import androidx.annotation.VisibleForTesting
+import androidx.core.app.ActivityCompat
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.OnLifecycleEvent
-import androidx.webkit.WebViewAssetLoader
 import com.rakuten.tech.mobile.miniapp.MiniAppDisplay
 import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
-import java.io.File
-
-private const val SUB_DOMAIN_PATH = "miniapp"
-private const val MINI_APP_INTERFACE = "MiniAppAndroid"
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 
 @SuppressLint("SetJavaScriptEnabled")
 internal class RealMiniAppDisplay(
-    context: Context,
+    val context: Context,
     val basePath: String,
     val appId: String,
-    miniAppMessageBridge: MiniAppMessageBridge
-) : MiniAppDisplay, WebView(context), WebViewListener {
-
-    private val miniAppDomain = "mscheme.$appId"
-    private val customScheme = "$miniAppDomain://"
-    private val customDomain = "https://$miniAppDomain/"
-
-    init {
-        layoutParams = FrameLayout.LayoutParams(
-            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT
-        )
+    val miniAppMessageBridge: MiniAppMessageBridge
+) : MiniAppDisplay {
 
-        settings.javaScriptEnabled = true
-        addJavascriptInterface(miniAppMessageBridge, MINI_APP_INTERFACE)
-        miniAppMessageBridge.setWebViewListener(this)
-
-        settings.allowUniversalAccessFromFileURLs = true
-        settings.domStorageEnabled = true
-        settings.databaseEnabled = true
-        webViewClient = MiniAppWebViewClient(context, getWebViewAssetLoader(), customDomain, customScheme)
+    @VisibleForTesting
+    internal var miniAppWebView: MiniAppWebView? = null
 
-        loadUrl(getLoadUrl())
-    }
+    override suspend fun getMiniAppView(): View? = provideMiniAppWebView(context)
```

Comment:
Actually, does this need to be a `suspend` function at all? It's just instantiating the `MiniAppWebView` class. `provideMiniAppWebView` is using the `Dispatchers.Main` context anyways and the App should already be calling it on the main thread since it returns a view. Is there any problem with removing the `suspend`?

Reply:
We need the `suspend` for `Dispatchers.Main`. `Dispatchers.Main` is required here for rendering the view because the hostapp can do `launch` with any `CoroutineScope` they want (not in main thread anymore).

For example, our sample app calling `launch {}` means launching with `Dispatchers.Default`. It's safer to specify `Dispatchers.Main`.